gateway/src/shard/error.rs

```diff
@@ -76,6 +77,19 @@ pub enum Error {
         /// The URL that couldn't be parsed.
         url: String,
     },
+    /// The payload received from Discord was an invalid structure.
+    ///
+    /// The payload was either invalid JSON or did not contain the necessary
+    /// "op" key in the object.
+    PayloadInvalid {
+        /// The payload received over the connection.
+        payload: String,
+    },
+    /// The binary payload received from Discord wasn't UTF-8 valid.
```

Comment:
Nit: `The binary payload received from Discord was not validly encoded as UTF-8.`